Tabloid/client/src/components/CategoryList.js

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useState } from "react";
+import { deleteCategory, getAllCategories } from "../modules/categoryManager";
+import { Link, useHistory } from "react-router-dom";
+
+export const CategoryList = () => {
+  const [categories, setCategories] = useState([]);
+
+  const getCategories = () => {
+    getAllCategories().then((categories) => setCategories(categories));
+  };
+
+  const history = useHistory();
+
+  useEffect(() => {
+    getCategories();
+  }, []);
+
+  const handleDelete = (id) => {
+    deleteCategory(id)
+      .then(getCategories())
```

Comment:
the .then method requires a callback function as its argument. Since you are invoking getCategories, and getCategories does not return anything, you're actually passing 'undefined'. you either need to pass the function by reference or add () => before you invoke getCategories here.

Reply:
Added () =>